pkg/controllers/jobset_controller.go

```diff
@@ -946,3 +968,91 @@ func managedByExternalController(js jobset.JobSet) *string {
 	}
 	return nil
 }
+
+func (r *JobSetReconciler) deleteJobSet(ctx context.Context, js *jobset.JobSet) error {
```

Comment:
can we please move all those functions to a `ttl_after_finished.go` file with its own unit tests file as well? like `startup_policy.go`?

this and the function below can receive the client and the clock as parameters instead of having the functions as members of jobsetReconciler.

Reply:
Should `deleteJobSet` function stay in the `jobset_controller.go`? It is a generic CRUD function, all of the other CRUD functions are in the same file

Reply:
It is generic, but only ttl after finished uses it, and likely will continue to be the only one